library/extractor/src/test/java/com/google/android/exoplayer2/extractor/mp4/Mp4ExtractorTest.java

```diff
@@ -101,5 +101,10 @@ public void mp4SampleWithMhm1Track() throws Exception {
   public void mp4SampleWithColorInfo() throws Exception {
     ExtractorAsserts.assertBehavior(
         Mp4Extractor::new, "media/mp4/sample_with_color_info.mp4", simulationConfig);
+
+  @Test
+  public void mp4SampleWithDolbyTrueHDTrack() throws Exception {
```

Comment:
I've messed up while rebasing. First a closing bracket } is  missing on line 104.

As of the regression on AC4, I have indeed broken it while moving most of the readSample method code into a readSample1 method.
I have now fixed both issues in an updated set of commits.

Reply:
The mp4SampleWithDolbyTrueHDTrack() test still fails when simulating i/o errors but I am not quite sure how to debug this because don't quite understand how i/o errors get injected.